src/main/java/appeng/container/implementations/ContainerInterfaceTerminal.java

```diff
@@ -407,30 +402,35 @@ private void addItems( final NBTTagCompound data, final InvTracker inv, final in
 		private final long sortBy;
 		private final long which = autoBase++;
 		private final String unlocalizedName;
-		private final IInventory client;
-		private final IInventory server;
+		private final IItemHandlerModifiable client;
+		private final IItemHandlerModifiable server;
 
-		public InvTracker( final DualityInterface dual, final IInventory patterns, final String unlocalizedName )
+		public InvTracker( final DualityInterface dual, final IItemHandlerModifiable patterns, final String unlocalizedName )
 		{
 			this.server = patterns;
-			this.client = new AppEngInternalInventory( null, this.server.getSizeInventory() );
+			this.client = new AppEngInternalInventory( null, this.server.getSlots() );
 			this.unlocalizedName = unlocalizedName;
 			this.sortBy = dual.getSortValue();
 		}
 	}
 
-	private static class PatternInvSlot extends WrapperInvSlot
+	private static class PatternSlotWrapper extends RangedWrapper
 	{
-
-		public PatternInvSlot( final IInventory inv )
-		{
-			super( inv );
+		public PatternSlotWrapper( final IItemHandlerModifiable inv, int slot )
+		{			
+			super( inv,  slot, slot + 1);
 		}
-
+		
 		@Override
-		public boolean isItemValid( final ItemStack itemstack )
-		{
-			return !itemstack.isEmpty() && itemstack.getItem() instanceof ItemEncodedPattern;
-		}
+		@Nonnull
+	    public ItemStack insertItem(int slot, @Nonnull ItemStack stack, boolean simulate)
+	    {
+			if (!stack.isEmpty() && stack.getItem() instanceof ItemEncodedPattern)
+			{
+	            return super.insertItem(slot, stack, simulate);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') '(' is not followed by whitespace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') ')' is not preceded with whitespace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)